donate/payments/views.py

```diff
@@ -0,0 +1,246 @@
+import logging
+import re
+import urllib
+
+from django.conf import settings
+from django.http import HttpResponseRedirect
+from django.template.defaultfilters import floatformat
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from django.views.generic import FormView, TemplateView
+
+from braintree import ErrorCodes
+
+from . import gateway
+from .exceptions import InvalidAddress
+from .forms import BraintreePaymentForm, PersonalDetailsForm
+from .utils import freeze_personal_details_for_session
+
+logger = logging.getLogger(__name__)
+
+
+class BraintreePaymentMixin(FormView):
```

Comment:
If this is subclassing a full-fledged view class like `FormView`, I would suggest going with `BraintreePaymentView` as a name here instead. A 'mixin' should be more agnostic about the other views it can be combined with.